source/Cosmos.System2/Graphics/VBECanvas.cs

```diff
@@ -44,6 +44,13 @@ public VBECanvas(Mode aMode)
         {
             Global.mDebugger.SendInternal($"Creating new VBEScreen() with mode {aMode.Columns}x{aMode.Rows}x{(uint)aMode.ColorDepth}");
 
+            if (Core.VBE.IsAvailable())
+            {
+                Core.VBE.ModeInfo ModeInfo = Core.VBE.getModeInfo();
+                aMode = new Mode(ModeInfo.width, ModeInfo.height, (ColorDepth)ModeInfo.bpp);
```

Comment:
Why are you overwriting the user selected mode?

Reply:
Because if VBE is enabled before Cosmos, the user can't change the mode (we need v8086 for that). So I get the current mode from Multiboot VBE ModeInfo